#! /bin/sh -e
## 50_g++-3.4.dpatch by  <rafael@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make rebuild.cc compile with g++ 3.4

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@

--- prcs-1.3.3.orig/src/rebuild.cc
+++ prcs-1.3.3/src/rebuild.cc
@@ -1001,10 +1001,14 @@
 #elif __GNUG__ == 3 and __GNUC_MINOR__ < 2
         buf = new filebuf(fdopen(dup(seg->fd()), "a+"), ios::out);
         buf->pubseekoff(0, ios::end, ios::out);
-#else
+#elif __GNUG__ == 3 and __GNUC_MINOR__ < 4
         buf = new __gnu_cxx::stdio_filebuf<char> (seg->fd(), ios::out,
 						  false /* close */, default_segment_size);
         buf->pubseekoff(0, ios::end, ios::out);
+#else
+        buf = new __gnu_cxx::stdio_filebuf<char> ((__c_file*)fdopen(seg->fd()) "w"), ios::out,
+						  default_segment_size);
+        buf->pubseekoff(0, ios::end, ios::out);
 #endif
 	os = new ostream(buf);
     }
